pkg/registry/apis/secret/service/secure_value.go

```diff
@@ -174,6 +174,10 @@ func (s *SecureValueService) createNewVersion(ctx context.Context, keeperName st
 		return nil, contracts.NewErrValidateSecureValue(errorList)
 	}
 
+	if sv.Spec.Ref != nil && keeperCfg.Type() == secretv1beta1.SystemKeeperType {
```

Comment:
This should be part of the validation as well no? if sv.Spec.Ref != nil && sv.Spec.Keeper == nil/system -> error

Reply:
Ah yeah 😄